test/helpers/envtest.go

```diff
@@ -56,6 +56,14 @@ func newTestEnvironment() *TestEnvironment {
 	_, filename, _, _ := goruntime.Caller(0) //nolint
 	root := path.Join(path.Dir(filename), "..", "..")
 
+	// Every time a CRD are created, ResourceVersion is set and this will prevent creation of
```

Comment:
@vincepri @benmoss in order to get tests passing I was forced to add this change to the test helper, that IMO is unrelated to current PR.
Without this change, tests were failing for me when creating more than one test environment in the same test session (e.g. this is happening for health check tests under controllers)...

Reply:
I've seen this as well. That said, can you explain in more details why setting the resource version on the CRD itself fixes this issue?

Reply:
The sequence was:
- the test suite creates a test environment
- create a test environment create CRDs objects
- when creating CRDs objects, each object gets updated with a ResourceVersion (CRDs are a pointer to an object)

Some tests, like health check tests, are creating another test environment instead of using the main one created by the suite
- create a test environment create CRDs objects
- when creating CRDs objects, the operation is failing because objects can't have the resourceVersion field set when passed to create

A possible alternative is to deep copy the CRD objects we are passing to the test environment

Reply:
I think we should probably move towards using `DeepCopy`, possibly by not even exposing these variables directly from `controllers/external`, but rather exposing a function that returns a deep copy instead. That way it's less error prone for consumption and we don't have to worry about possible effects during parallel runs if the resource is changed in other ways other than the resource version.

Reply:
moved to func

Reply:
rolled back to DeepCopy because it is a breaking change